lib/modules/actions.js

```diff
@@ -54,7 +56,7 @@ export const reroutePage = () => async (dispatch, getState) => {
       queryParams: currentPage.queryParams,
       hashParams: currentPage.hashParams,
       bodyParams: {},
-      referrer: currentPage.referrer,
+      referrer: currentPage.referrer || NO_REFERRER,
```

Comment:
Are you sure? In any case, on direct navigation currentPage.referrer is an empty string which is exactly what it is when the user changes routes on the client making them indistinguishable. We need to be able to distinguish between the two.